frontend/src/component/pages/shared/CreateShortLinkSection.tsx

```diff
@@ -7,77 +7,188 @@ import { ShortLinkUsage } from './ShortLinkUsage';
 import { Section } from '../../ui/Section';
 import { Url } from '../../../entity/Url';
 import { UIFactory } from '../../UIFactory';
+import { validateLongLinkFormat } from '../../../validators/LongLink.validator';
+import { validateCustomAliasFormat } from '../../../validators/CustomAlias.validator';
+import { raiseCreateShortLinkError } from '../../../state/actions';
+import { IAppState } from '../../../state/reducers';
+import { Store } from 'redux';
+import { UrlService } from '../../../service/Url.service';
+import { QrCodeService } from '../../../service/QrCode.service';
 
-interface Props {
+interface IProps {
+  store: Store<IAppState>;
   uiFactory: UIFactory;
-  longLinkText?: string;
+  urlService: UrlService;
+  qrCodeService: QrCodeService;
+  onShortLinkCreated?: (shortLink: string) => void;
+  onRequestSignIn?: () => void;
```

Comment:
I would prefer it being named as `authenticationErrHandler`. Let the parent component decide what to do.

Reply:
Done.